### PR TITLE
modify tke config and new cos bucket policy file

### PR DESCRIPTION
**Pull request by cyberHermanwang**
--tke resource 
make testacc TESTARGS="-run=TestAccTencentCloudTkeAsResource"  
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudTkeAsResource -timeout 120m
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/gendoc	0.680s [no tests to run]
=== RUN   TestAccTencentCloudTkeAsResource
--- PASS: TestAccTencentCloudTkeAsResource (320.40s)
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud	321.034s
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]
--cos bucket policy resource
 make testacc TESTARGS="-run=TestAccTencentCloudCosBucketPolicy"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudCosBucketPolicy -timeout 120m
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]
=== RUN   TestAccTencentCloudCosBucketPolicy
--- PASS: TestAccTencentCloudCosBucketPolicy (9.45s)
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud	10.211s
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]

### CODE REVIEW

**Comment by cyberHermanwang** [comment]
make testacc TESTARGS="-run=TestAccTencentCloudCosBucketPolicy"       
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudCosBucketPolicy -timeout 120m
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/gendoc	0.896s [no tests to run]
=== RUN   TestAccTencentCloudCosBucketPolicy
--- PASS: TestAccTencentCloudCosBucketPolicy (10.87s)
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud	11.629s
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,13 @@
 ## 1.48.1 (Unreleased)
```
Comment (gailwang) [inline CHANGELOG.md:1]: New resource  1.49

File: examples/tencentcloud-cos/variables.tf [hunk 2]
```diff
@@ -13,3 +13,28 @@ variable "acl" {
...
+          "qcs::cam::uin/100010835595:uin/100014918835"
+        ]
+      },
+      "Action": [
+        "name/cos:DeleteBucket",
+        "name/cos:PutBucketACL"
+      ],
+      "Effect": "allow",
+      "Resource": [
+        "qcs::cos:ap-nanjing:uid/1259649581:hhermanwang-1259649581/*"
```
Comment (gailwang) [inline examples/tencentcloud-cos/variables.tf:34]: example里不要带自己的信息

File: tencentcloud/resource_tc_clb_listener.go [hunk 3]
```diff
@@ -352,6 +352,7 @@ func resourceTencentCloudClbListener() *schema.Resource {
 				Type:         schema.TypeString,
 				Optional:     true,
 				ForceNew:     true,
+				Computed:     true,
```
Comment (gailwang) [inline tencentcloud/resource_tc_clb_listener.go:355]: 这里为什么是computed？

File: tencentcloud/resource_tc_cos_bucket_policy.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+					}
+					var newJson interface{}
+					err = json.Unmarshal([]byte(news), &newJson)
+					if err != nil {
+						return olds == news
+					}
+					flag := reflect.DeepEqual(oldJson, newJson)
+					return flag
+				},
+				Description: "The text of the policy. this field is required. the syntax refers to https://cloud.tencent.com/document/product/436/18023.",
```
Comment (gailwang) [inline tencentcloud/resource_tc_cos_bucket_policy.go:93]: This， 找一下markdown里文档超链接的写法

Comment (gailwang) [reply tencentcloud/resource_tc_cos_bucket_policy.go:93]: 文档不要给中文的，This is required不需要。Required在前面展示了，可以看一下生成的doc文件

File: tencentcloud/resource_tc_clb_listener.go (hunk 3 above)
Comment (cyberHermanwang) [reply tencentcloud/resource_tc_clb_listener.go:355]: 这个是帮俊楚提的一行代码

File: tencentcloud/resource_tc_kubernetes_as_scaling_group.go [hunk 5]
```diff
@@ -880,6 +878,45 @@ func resourceKubernetesAsScalingGroupCreate(d *schema.ResourceData, meta interfa
...
+		items   = strings.Split(d.Id(), ":")
+	)
+	if len(items) != 2 {
+		return fmt.Errorf("resource_tc_kubernetes_as_scaling_group id  is broken")
+	}
+	clusterId := items[0]
+	asGroupId := items[1]
+	if d.HasChange("auto_scaling_group") {
+		asGroups := d.Get("auto_scaling_group").([]interface{})
+		if len(asGroups) != 1 {
```
Comment (gailwang) [inline tencentcloud/resource_tc_kubernetes_as_scaling_group.go:897]: 关于这个校验 上面有一个Required + MaxItem=1，应该是不需要的

**Review by gailwang — APPROVED** [review]

**Review by crab21 — APPROVED** [review]

**Pull request merged**